components/popover/popover-mixin.js

```diff
@@ -62,34 +69,105 @@ export const PopoverMixin = superclass => class extends superclass {
 
 	constructor() {
 		super();
+		this.noAutoClose = false;
 		this.opened = false;
-		this._useNativePopover = isSupported;
+		this._useNativePopover = isSupported ? 'manual' : undefined;
+		this._handleAutoCloseClick = this._handleAutoCloseClick.bind(this);
+		this._handleAutoCloseFocus = this._handleAutoCloseFocus.bind(this);
 	}
 
 	connectedCallback() {
 		super.connectedCallback();
+		if (this.opened) this._addAutoCloseHandlers();
 	}
 
 	disconnectedCallback() {
 		super.disconnectedCallback();
+		this._removeAutoCloseHandlers();
 	}
 
 	updated(changedProperties) {
 		super.updated(changedProperties);
 		if (changedProperties.has('opened')) {
 
+			if (this._useNativePopover) {
+				if (this.opened) this.showPopover();
+				else this.hidePopover();
+			}
+
+			this._previousFocusableAncestor = this.opened ? getPreviousFocusableAncestor(this, false, false) : null;
+
 			if (this.opened) {
+				this._opener = getComposedActiveElement();
+				this._addAutoCloseHandlers();
 				this.dispatchEvent(new CustomEvent('d2l-popover-open', { bubbles: true, composed: true }));
 			} else if (changedProperties.get('opened') !== undefined) {
+				this._removeAutoCloseHandlers();
 				this.dispatchEvent(new CustomEvent('d2l-popover-close', { bubbles: true, composed: true }));
 			}
 
-			if (this._useNativePopover) {
-				if (this.opened) this.showPopover();
-				else this.hidePopover();
-			}
+		}
+	}
+
+	_addAutoCloseHandlers() {
+		this.addEventListener('blur', this._handleAutoCloseFocus, { capture: true });
+		document.body.addEventListener('focus', this._handleAutoCloseFocus, { capture: true });
+		document.addEventListener('click', this._handleAutoCloseClick, { capture: true });
+	}
+
+	_close() {
+		const hide = () => {
+			this.opened = false;
+		};
+
+		hide();
+	}
+
+	_handleAutoCloseClick(e) {
+
+		if (!this.opened || this.noAutoClose) return;
 
+		const rootTarget = e.composedPath()[0];
+		if (isComposedAncestor(this.shadowRoot.querySelector('.content'), rootTarget)
+			|| isComposedAncestor(this._opener, rootTarget)) {
+			return;
 		}
+
+		this._close();
+	}
+
+	_handleAutoCloseFocus() {
+
+		// timeout needed to work around lack of support for relatedTarget
+		setTimeout(() => {
+			// we ignore focusable ancestors othrwise the popover will close when user clicks empty space inside the popover
+			if (!this.opened
+				|| this.noAutoClose
+				|| !document.activeElement
+				|| document.activeElement === this._previousFocusableAncestor
+				|| document.activeElement === document.body) {
+				return;
+			}
+
+			const activeElement = getComposedActiveElement();
+			if (isComposedAncestor(this.shadowRoot.querySelector('.content'), activeElement)
+				|| activeElement === this._opener) {
+				return;
+			}
+
+			this._close();
+		}, 0);
+
+	}
+
+	async _open() {
```

Comment:
Does this need to be async? Or is that something that will be needed?

Reply:
Ah yep, this will be needed. I tried to clean up the code to only include what was relevant for this PR but I missed this.

Reply:
Removed this method to keep the commits cleaner.